### ISSUE TITLE
SNAPSHOT artifacts are missing `alpha` in the name.

### ISSUE THREAD

**Issue by breedx-splk** [open-telemetry/opentelemetry-android#275]
Maybe we don't care too much since you already have to jump through hoops to use the SNAPSHOT builds, but there is definitely an inconsistency. I think we should include `alpha` in the SNAPSHOT filenames as well.

Here's an example:
https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/android/agent/maven-metadata.xml

```xml

<metadata>
  <groupId>io.opentelemetry.android</groupId>
  <artifactId>agent</artifactId>
  <versioning>
    <latest>0.5.0-SNAPSHOT</latest>
    <versions>
      <version>0.5.0-SNAPSHOT</version>
    </versions>
    <lastUpdated>20240312081534</lastUpdated>
  </versioning>
</metadata>
```

**Comment by breedx-splk** [comment]
Hmmm, maybe this isn't still an issue, just an artifact of the multi-step project reorg?
https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/android/android-agent/

**Comment by marandaneto** [comment]
sonatype should not have SNAPSHOT artifacts, they have a different repository for that.
https://central.sonatype.org/publish/publish-guide/#accessing-repositories

what I usually do/did is that I don't publish snapshots in the snapshot repository either but I publish semver builds in the stable repository with suffixes (alpha, beta, RC, stable).
Also, check this recommendation for the suffixes https://publicobject.com/2019/12/18/naming-versions/ otherwise you might get the wrong version as the latest stable version (sorting issue).

**Comment by LikeTheSalad** [comment]
I think this only happens to all the modules except for the `android-agent` one, because the logic to append "-alpha" is added only there: https://github.com/open-telemetry/opentelemetry-android/blob/783ef1b8ae48ecbf4d0c4226c0fa35fe220da2b4/android-agent/build.gradle.kts#L8 

So I think it should be a matter of moving that logic to the convention scripts to cover all other modules as well.

**Comment by breedx-splk** [comment]
Fixed in #297
